actionpack/lib/action_controller/test_case.rb

```diff
@@ -208,10 +207,14 @@ def load!
   end
 
   # Superclass for ActionController functional tests. Functional tests allow you to
-  # test a single controller action per test method. This should not be confused with
-  # integration tests (see ActionDispatch::IntegrationTest), which are more like
-  # "stories" that can involve multiple controllers and multiple actions (i.e. multiple
-  # different HTTP requests).
+  # test a single controller action per test method.
+  #
+  # == Use `ActionDispatch::Integration` tests instead of this.
+  #
+  # Rails discourages use of `ActionController` functional tests in favor of integration tests
+  # (see ActionDispatch::IntegrationTest).
+  # New Rails applications no longer generate `ActionController` style functional tests and they should
```

Comment:
As a reader, I'm more interested in the why than what the Rails generators do. We've moved to integration tests because they perform a request including executing the middleware stack and encoding the parameters, whereas functional controller tests merely simulate a request. Plus integration tests are faster and support `as: :json` and `parsed_body` to help ease API testing. Basically, I think it's a good thing to try to nudge people over to integration tests as best we can 😁